src/libraries/PCPricing.sol

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+/// @title Preferred Currency Pricing (PCPricing) Library
+/// @author Permissionless Games & ChatGpt
+/// @notice This library enables dynamic price adjustments based on preferred currency usage.
+/// @dev When the anchor currency is used, the price of all other tracked currencies can be decreased.
+///      A universal adjustment factor applies to all non-anchor currencies, promoting dynamic pricing models.
+
+library PCPricing {
+    using PCPricing for PCPricing.PricingData;
+    /// @notice Emitted when a new anchor currency is set
+    /// @param currency The currency that was set as the anchor
+    /// @param price The price of the anchor currency
+
+    event AnchorCurrencySet(bytes indexed currency, uint256 price);
```

Comment:
Explain how currencies are encoded as bytes in the comments.

Reply:
Add `encodeCurrency` and `decodeCurrency` methods. Or should this not be a concern of the library?